File: scaffold/src/main/java/top/xuqingquan/web/agent/IndicatorController.java
package top.xuqingquan.web.agent;

public interface IndicatorController {

    void progress(android.webkit.WebView v, int newProgress);

    void progress(com.tencent.smtt.sdk.WebView v, int newProgress);

    BaseIndicatorSpec offerIndicator();

    void showIndicator();

    void setProgress(int newProgress);

    void finish();
}

File: scaffold/src/main/java/top/xuqingquan/web/agent/IndicatorController.kt
package top.xuqingquan.web.agent

/**
 * Created by 许清泉 on 2019-06-05 00:53
 */
interface IndicatorController {

    fun offerIndicator(): BaseIndicatorSpec

    fun showIndicator()

    fun setProgress(newProgress: Int)

    fun finish()
}